### DISCUSSION TITLE
Cannot edit response object - Object frozen

### DISCUSSION THREAD

**Discussion by iPOiZN** [nuxt-community/apollo-module#613] [Q&A]
Hi, I have the following code:
const GetChatsQuery = gql  `query($withBlocked: Boolean) {  userChats(withBlocked: $withBlocked) {  ${ GetChatGraphQL(10) } }  };`

    ` return this.apollo.query({
        query: GetChatsQuery,
        variables: {
            withBlocked: withBlocked
        }
    }).map(res => {
        let response = <GetChatsResponse>res.data;
        // I get here the error
        response.userChats[0].messages.push(null);
        return response.userChats;
    })`
The thing is that I'm getting an error on the push method:

Cannot add property 2, object is not extensible

As I understood from the internet it means that the object is frozen (Object.freeze).
Have anyone had that problem before? how can I solve it?

**Comment by mpgalaxy** [comment]
Hi, yes I had the same problem, solved it by creating a copy of the result `JSON.parse(JSON.stringify(res.data))`. This won't be blocked. Kind of dirty, but I couldn't find a better solution. ;)
